### PR TITLE
New: switch-colon-spacing rule (fixes #7981)

### LINKED ISSUE #7981: Disallow space before colon in SwitchStatement
Hi guys!
Today there is no way to disallow space before colon in switch-case statement:
```js
switch (foo) {
  case 'bar' : 
    return '^---oh, that space...';
}
```
I already use `keyword-spacing`, but I can't configure it for my case. I tried `key-spacing` too, but it still doesn't work.
My configuration looks like this:
```
    ...
    "key-spacing": "error",
    "keyword-spacing": "error",
    ...
```
You can also reproduce this by http://eslint.org/demo/ with enabled `keyword-spacing` & `key-spacing`.

I've asked about this in gitter and I was recommended to create an issue. I don't know, should it be a new rule or modification of an existing?

Thank you!

### PR DESCRIPTION
**Pull request by mysticatea**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))

Fixes #7981.

**What changes did you make? (Give an overview)**

This PR add new rule `switch-colon-spacing` which enforces spacing style around colons of switch statements.

**Is there anything you'd like reviewers to focus on?**

Please check whether the tests and the document are enough.

### CODE REVIEW

File: lib/rules/switch-colon-spacing.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+/**
+ * @fileoverview Rule to
```
Comment (mysticatea) [inline lib/rules/switch-colon-spacing.js]: Oh, I have forgotten it. The line was from my template.

File: docs/rules/switch-colon-spacing.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Enforce spacing around colons of switch statements (switch-colon-spacing)
+
+Spacing around colons improves readability of `case`/`default` closures.
+
+## Rule Details
+
+This rule reports around colons of `case`/`default` closures if it's different to configured spacing.
```
Comment (btmills) [inline docs/rules/switch-colon-spacing.md]: I would rephrase this to "This rule controls spacing around colons of `case` and `default` clauses in `switch` statements."

File: docs/rules/switch-colon-spacing.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Enforce spacing around colons of switch statements (switch-colon-spacing)
+
+Spacing around colons improves readability of `case`/`default` closures.
```
Comment (btmills) [inline docs/rules/switch-colon-spacing.md]: Is "closure" the right word here? I've more often heard these called "clauses".
